krm-functions/nfdeploy-fn/mutator/mutator.go

```diff
@@ -0,0 +1,168 @@
+package mutator
+
+import (
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	nephiodeployv1alpha1 "github.com/nephio-project/api/nf_deployments/v1alpha1"
+	nephioreqv1alpha1 "github.com/nephio-project/api/nf_requirements/v1alpha1"
+	infrav1alpha1 "github.com/nephio-project/nephio-controller-poc/apis/infra/v1alpha1"
+	kptcondsdk "github.com/nephio-project/nephio/krm-functions/lib/condkptsdk"
+	corev1 "k8s.io/api/core/v1"
+	"reflect"
+)
+
+func Run(rl *fn.ResourceList) (bool, error) {
+	nfDeployFn := NewMutator()
+
+	var err error
+
+	nfDeployFn.sdk, err = kptcondsdk.New(
+		rl,
+		&kptcondsdk.Config{
+			For: corev1.ObjectReference{
+				APIVersion: nephiodeployv1alpha1.GroupVersion.Identifier(),
+				Kind:       nephiodeployv1alpha1.UPFDeploymentKind,
+			},
+			Watch: map[corev1.ObjectReference]kptcondsdk.WatchCallbackFn{
+				{
+					APIVersion: infrav1alpha1.GroupVersion.Identifier(),
+					Kind:       reflect.TypeOf(infrav1alpha1.ClusterContext{}).Name(),
+				}: nfDeployFn.ClusterContextCallBackFn,
+				{
+					APIVersion: nephioreqv1alpha1.GroupVersion.Identifier(),
+					Kind:       reflect.TypeOf(nephioreqv1alpha1.Capacity{}).Name(),
+				}: nfDeployFn.CapacityContextCallBackFn,
+				{
+					APIVersion: nephioreqv1alpha1.GroupVersion.Identifier(),
+					Kind:       nephioreqv1alpha1.InterfaceKind,
+				}: nfDeployFn.InterfaceCallBackFn,
+				{
+					APIVersion: nephioreqv1alpha1.GroupVersion.Identifier(),
+					Kind:       nephioreqv1alpha1.DataNetworkKind,
+				}: nfDeployFn.DnnCallBackFn,
+				// TODO: Should also include Kptfile for UPF Deployment Name
+			},
+			GenerateResourceFn: nfDeployFn.GenerateResourceFn,
+		},
+	)
+
+	if err != nil {
+		rl.Results.ErrorE(err)
+		return false, err
+	}
+
+	nfDeployFn.sdk.Run()
+	return true, nil
+}
+
+func (h *NfDeployFn) ClusterContextCallBackFn(o *fn.KubeObject) error {
+	var cluster infrav1alpha1.ClusterContext
+	err := o.As(&cluster)
+	if err != nil {
+		return err
+	}
+	h.site = *cluster.Spec.SiteCode
+	return nil
+}
+
+func (h *NfDeployFn) CapacityContextCallBackFn(o *fn.KubeObject) error {
+	var capacity nephioreqv1alpha1.Capacity
+	err := o.As(&capacity)
+	if err != nil {
+		return err
+	}
+
+	h.SetCapacity(capacity.Spec.MaxDownlinkThroughput, capacity.Spec.MaxUplinkThroughput)
+	return nil
+}
+
+func (h *NfDeployFn) InterfaceCallBackFn(o *fn.KubeObject) error {
+	var itfce *nephioreqv1alpha1.Interface
+	err := o.As(&itfce)
+	if err != nil {
+		return err
+	}
+
+	if itfce.Status.IPAllocationStatus == nil || itfce.Status.VLANAllocationStatus == nil {
+		return nil
+	}
+
+	itfcIPAllocStatus := itfce.Status.IPAllocationStatus
```

Comment:
May be better to move this above the nil check and use the same variables throughout